### PR TITLE
Revert: Define Propagation-only Span to simplify active Span logic in Context

### LINKED ISSUE #1124: Revert use of PropagatedSpan back to using a special Context key for extracted Span Context
**What are you trying to achieve?**

In discussion last night in the Spec SIG meeting we talked about the confusion added with PropagatedSpan and whether it didn't cause more confusion than the issue it was trying to resolve (the code required to check if an extracted Span Context exists if there isn't a parent in the current Context or the Context manually passed as the parent to `StartSpan`.

I'm opening a revert of the PropagatedSpan PR to go along with this issue.

### PR DESCRIPTION
**Pull request by tsloughter**
This reverts commit d14c9b7a74328c22e5d2223c4ddcaca5630aceae.

Fixes #1124

### CODE REVIEW

**Comment by bogdandrutu** [comment]
@tsloughter can we document the reasons?

File: specification/trace/api.md [hunk 1]
```diff
@@ -32,7 +32,11 @@ Table of Contents
     * [End](#end)
     * [Record Exception](#record-exception)
   * [Span lifetime](#span-lifetime)
-  * [Propagated Span creation](#propagated-span-creation)
+* [Status](#status)
+  * [StatusCanonicalCode](#statuscanonicalcode)
```
Comment (jkwatson) [inline specification/trace/api.md:36]: This can't be right. I think your revert is reverting too much.

Comment (tsloughter) [reply specification/trace/api.md:36]: Oh, damn, yea. There were some conflicts I had to resolve and must have got this one wrong, though I thought the other part of the diff was empty.. but I'll get this fixed.

**Review by Oberon00 — CHANGES_REQUESTED** [review]
Blocking this for now -- please see discussion on issue.

**Pull request closed without merging**